packages/plugin-nft-generation/src/provider/wallet/walletSolana.ts

```diff
@@ -0,0 +1,231 @@
+            });
+        }
+        const umi = createUmi(this.connection.rpcEndpoint);
+        umi.use(mplTokenMetadata());
+        const umiUser = umi.eddsa.createKeypairFromSecretKey(
+            this.privateKeyUint8Array
+        );
+        umi.use(keypairIdentity(umiUser));
+        this.umi = umi;
+    }
+
+    async getBalance() {
+        let balance = await this.connection.getBalance(this.walletPublicKey);
+        return {
+            value: balance,
+            formater: `${balance / LAMPORTS_PER_SOL} SOL`,
+        };
+    }
+
+    get privateKeyUint8Array() {
+        return bs58.decode(this.walletPrivateKeyKey);
+    }
+
+    async createCollection({
+        name,
+        symbol,
+        adminPublicKey,
+        uri,
+        fee,
+    }: {
+        name: string;
+        symbol: string;
+        adminPublicKey: string;
+        uri: string;
+        fee: number;
+    }): Promise<any> {
+        const collectionMint = generateSigner(this.umi);
+        elizaLogger.log("collectionMint", collectionMint, percentAmount(5));
+        let transaction = new TransactionBuilder();
+        const info = {
+            name,
+            symbol,
+            uri,
+        };
+        console.log(`Metadata uploaded: ${uri}`);
+        transaction = transaction.add(
+            createNft(this.umi, {
+                ...info,
+                mint: collectionMint,
+                sellerFeeBasisPoints: percentAmount(fee),
+                isCollection: true,
+            })
+        );
+
+        transaction = transaction.add(
+            updateV1(this.umi, {
+                mint: collectionMint.publicKey,
+                newUpdateAuthority: publicKey(adminPublicKey), // updateAuthority's public key
+            })
+        );
+
+        await transaction.sendAndConfirm(this.umi, {
+            confirm: {},
+        });
+
+        // await sleep(5000);
+        // const createdCollectionNft = await fetchDigitalAsset(
+        //     this.umi,
+        //     collectionMint.publicKey
+        // );
+        const address = collectionMint.publicKey;
+        return {
+            link: getExplorerLink("address", address, this.cluster),
+            address,
+        };
+    }
+
+    async mintNFT({
+        collectionAddress,
+        adminPublicKey,
+        name,
+        symbol,
+        uri,
+        fee,
+    }: {
+        collectionAddress: string;
+        adminPublicKey: string;
+        name: string;
+        symbol: string;
+        uri: string;
+        fee: number;
+    }): Promise<any> {
+        const umi = this.umi;
+        const mint = generateSigner(umi);
+
+        let transaction = new TransactionBuilder();
+        console.log("collectionAddress", collectionAddress);
+        const collectionAddressKey = publicKey(collectionAddress);
+        // Add SOL transfer instruction (0.1 SOL)
+        const receiverAddressKey = publicKey(adminPublicKey); // Replace with actual receiver address
```

Comment:
This says `Replace with actual receiver address`. What is the implication of having the `adminPublicKey` here for now?

Reply:
The adminPublicKey is used to verify the ownership relationship between NFTs and collections. Here is debugging a feature: verify after receiving the payment.
This debugging code is not necessary, I will remove it.